common/src/verify.rs

```diff
@@ -1450,5 +1470,87 @@ mod test {
         .unwrap_err();
     }
 
-    // TODO: add test case where extra-agenda transactions are invalid.
+    #[ignore]
+    #[test]
+    // Test the case where the `Delegate` extra-agenda transaction is invalid because the delegator is not a member.
+    /// This test case is ignored because the extra-agenda transaction is not implemented yet.
+    // TODO: enable this test case when the extra-agenda transaction is implemented.
+    fn invalid_delegate_transaction_with_invalid_delegator1() {
+        todo!("Implement this test")
+    }
+
+    #[ignore]
+    #[test]
+    // Test the case where the `Delegate` extra-agenda transaction is invalid because the delegator has already delegated.
+    /// This test case is ignored because the extra-agenda transaction is not implemented yet.
+    // TODO: enable this test case when the extra-agenda transaction is implemented.
+    fn invalid_delegate_transaction_with_invalid_delegator2() {
+        todo!("Implement this test")
+    }
+
+    #[ignore]
+    #[test]
+    // Test the case where the `Delegate` extra-agenda transaction is invalid because the delegatee is not a member.
+    /// This test case is ignored because the extra-agenda transaction is not implemented yet.
+    // TODO: enable this test case when the extra-agenda transaction is implemented.
+    fn invalid_delegate_transaction_with_invalid_delegatee() {
+        todo!("Implement this test")
+    }
+
+    #[ignore]
+    #[test]
+    // Test the case where the `Delegate` extra-agenda transaction is invalid because the signature is invalid.
+    /// This test case is ignored because the extra-agenda transaction is not implemented yet.
+    // TODO: enable this test case when the extra-agenda transaction is implemented.
+    fn invalid_delegate_transaction_with_invalid_signature() {
+        todo!("Implement this test")
+    }
+
+    #[ignore]
+    #[test]
+    // Test the case where the `Delegate` extra-agenda transaction is invalid because the timestamp is invalid.
+    /// This test case is ignored because the extra-agenda transaction is not implemented yet.
+    // TODO: enable this test case when the extra-agenda transaction is implemented.
+    fn invalid_delegate_transaction_with_invalid_timestamp() {
+        todo!("Implement this test")
+    }
+
+    #[ignore]
+    #[test]
+    // Test the case where the `Undelegate` extra-agenda transaction is invalid because the delegator is not a member.
+    /// This test case is ignored because the extra-agenda transaction is not implemented yet.
+    /// TODO: enable this test case when the extra-agenda transaction is implemented.
+    fn invalid_undelegate_transaction_with_invalid_delegator1() {
+        todo!("Implement this test")
+    }
+
+    #[ignore]
+    #[test]
+    // Test the case where the `Undelegate` extra-agenda transaction is invalid because the delegator has nothing delegated.
```

Comment:
I can't understand the meaning of `has nothing delegated`. Can you explain or change this?

Reply:
It means the delegator has not been delegated with any voting power. Is this more clear with you? I think the English is correct. But, I'll fix which is more clear to you.

@junha1 Can you also give a comment on this please?

Reply:
I think the point here is that the designated member is not even a 'delegator', which's quite confusing. How about "because the member is not a delegator" or so?

Reply:
@junha1 Really? Acutally that case is the one above `invalid_undelegate_transaction_with_invalid_delegator1`.
I think it's just the expression that might be not that clear enough.

Reply:
@TomTaehoonKim You mean the delegator has never used the voting power in this case?